app/src/pages/inside/dashboardPage/dashboardPage.jsx

```diff
@@ -1,4 +1,228 @@
+  pageTitle: {
+    id: 'DashboardPage.title',
+    defaultMessage: 'All Dashboards',
+  },
+  modalCancelButtonText: {
+    id: 'DashboardPage.modal.modalCancelButtonText',
+    defaultMessage: 'Cancel',
+  },
+  deleteModalWarningMessage: {
+    id: 'DashboardPage.modal.deleteModalWarningMessage',
+    defaultMessage:
+      'You are going to delete not your own dashboard. This may affect other users information on the project.',
+  },
+  deleteModalTitle: {
+    id: 'DashboardPage.modal.deleteModalTitle',
+    defaultMessage: 'Delete Dashboard',
+  },
+  deleteModalConfirmationText: {
+    id: 'DashboardPage.modal.deleteModalConfirmationText',
+    defaultMessage: 'Are you sure to delete dashboard {name}? It will no longer exist.',
+  },
+  deleteModalSubmitButtonText: {
+    id: 'DashboardPage.modal.deleteModalSubmitButtonText',
+    defaultMessage: 'Delete',
+  },
+  editModalTitle: {
+    id: 'DashboardPage.modal.editModalTitle',
+    defaultMessage: 'Edit Dashboard',
+  },
+  editModalSubmitButtonText: {
+    id: 'DashboardPage.modal.editModalSubmitButtonText',
+    defaultMessage: 'Update',
+  },
+  addModalTitle: {
+    id: 'DashboardPage.modal.addModalTitle',
+    defaultMessage: 'Add New Dashboard',
+  },
+  addModalSubmitButtonText: {
+    id: 'DashboardPage.modal.addModalSubmitButtonText',
+    defaultMessage: 'Add',
+  },
+});
+
+@connect(
+  (state) => ({
+    gridType: dashboardGridTypeSelector(state),
+    dashboardItems: dashboardItemsSelector(state),
+    userInfo: userInfoSelector(state),
+  }),
+  {
+    fetchDashboard: fetchDashboardAction,
+    changeVisibilityType: changeVisibilityTypeAction,
+    showModal: showModalAction,
+    deleteDashboard: deleteDashboardAction,
+    editDashboard: editDashboardAction,
+    addDashboard: addDashboardAction,
+  },
+)
+@withFilter
+@injectIntl
+export class DashboardPage extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    showModal: PropTypes.func,
+    deleteDashboard: PropTypes.func,
+    editDashboard: PropTypes.func,
+    addDashboard: PropTypes.func,
+    fetchDashboard: PropTypes.func,
+    userInfo: PropTypes.object,
+    filter: PropTypes.string,
+    dashboardItems: PropTypes.array,
+    gridType: PropTypes.string,
+    onFilterChange: PropTypes.func,
+    changeVisibilityType: PropTypes.func,
+  };
+
+  static defaultProps = {
+    showModal: () => {},
+    deleteDashboard: () => {},
+    editDashboard: () => {},
+    addDashboard: () => {},
+    fetchDashboard: () => {},
+    userInfo: {},
+    filter: '',
+    dashboardItems: [],
+    gridType: '',
+    onFilterChange: () => {},
+    changeVisibilityType: () => {},
+  };
+
+  componentDidMount() {
+    const { fetchDashboard, changeVisibilityType } = this.props;
+
+    fetchDashboard();
+    changeVisibilityType();
+  }
+
+  onDeleteDashboardItem = (item) => {
+    const { showModal, deleteDashboard, userInfo: { userId }, intl } = this.props;
+
+    showModal({
+      id: 'dashboardModal',
```

Comment:
Why don't we just show different modals instead of `dashboardModal` with some `type` parameter? For example, in this case it would be something like `dashboardDeleteModal`